lib/ex_phone_number/metadata/phone_number_description.ex

```diff
@@ -9,28 +9,70 @@ defmodule ExPhoneNumber.Metadata.PhoneNumberDescription do
   def from_xpath_node(nil), do: nil
   def from_xpath_node(xpath_node) do
     kwlist =
-      xpath_node |> xmap(
-        national_number_pattern: ~x"./nationalNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
-        possible_number_pattern: ~x"./possibleNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
-        example_number: ~x"./exampleNumber/text()"o |> transform_by(&normalize_string/1)
-      )
-    struct(%PhoneNumberDescription{}, kwlist)
+      xpath_node
+      |> xmap(
+           national_number_pattern: ~x"./nationalNumberPattern/text()"o |> transform_by(&normalize_pattern/1),
+           national_possible_lengths: ~x"./possibleLengths/@national"o |> transform_by(&normalize_range/1),
+           local_possible_lengths: ~x"./possibleLengths/@localOnly"o |> transform_by(&normalize_range/1),
+           example_number: ~x"./exampleNumber/text()"o |> transform_by(&normalize_string/1)
+         )
+
+    possible_lengths = case is_list(kwlist.local_possible_lengths) do
```

Comment:
Maybe:
```elixir
possible_lengths =
  (kwlist.local_possible_lengths || [])
  |> Enum.join(kwlist.national_possible_lengths || [])
  |> Enum.dedup()
```

Reply:
`concat` instead of `join` but yes ! Much nicer way to write it :)